tests/components/youtube/test_config_flow.py

```diff
@@ -0,0 +1,248 @@
+"""Test the YouTube config flow."""
+from unittest.mock import patch
+
+from httplib2 import Response
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.youtube.const import CONF_CHANNELS, DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from . import MockService
+from .conftest import (
+    CLIENT_ID,
+    GOOGLE_AUTH_URI,
+    GOOGLE_TOKEN_URI,
+    SCOPES,
+    TITLE,
+    ComponentSetup,
+)
+
+from tests.common import MockConfigEntry, load_fixture
+from tests.test_util.aiohttp import AiohttpClientMocker
+from tests.typing import ClientSessionGenerator
+
+
+async def test_full_flow(
+    hass: HomeAssistant,
+    hass_client_no_auth: ClientSessionGenerator,
+    current_request_with_host: None,
+) -> None:
+    """Check full flow."""
+    result = await hass.config_entries.flow.async_init(
+        "youtube", context={"source": config_entries.SOURCE_USER}
+    )
+    state = config_entry_oauth2_flow._encode_jwt(
+        hass,
+        {
+            "flow_id": result["flow_id"],
+            "redirect_uri": "https://example.com/auth/external/callback",
+        },
+    )
+
+    assert result["url"] == (
+        f"{GOOGLE_AUTH_URI}?response_type=code&client_id={CLIENT_ID}"
+        "&redirect_uri=https://example.com/auth/external/callback"
+        f"&state={state}&scope={'+'.join(SCOPES)}"
+        "&access_type=offline&prompt=consent"
+    )
+
+    client = await hass_client_no_auth()
+    resp = await client.get(f"/auth/external/callback?code=abcd&state={state}")
+    assert resp.status == 200
+    assert resp.headers["content-type"] == "text/html; charset=utf-8"
+
+    with patch(
+        "homeassistant.components.youtube.async_setup_entry", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.youtube.api.build", return_value=MockService()
+    ), patch(
+        "homeassistant.components.youtube.config_flow.build", return_value=MockService()
+    ):
+        result = await hass.config_entries.flow.async_configure(result["flow_id"])
+        assert result["type"] == FlowResultType.FORM
+        assert result["step_id"] == "channels"
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={CONF_CHANNELS: ["UC_x5XG1OV2P6uZZ5FSM9Ttw"]}
+        )
+
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+    assert len(mock_setup.mock_calls) == 1
+
+    assert result.get("type") == "create_entry"
+    assert result.get("title") == TITLE
+    assert "result" in result
+    assert result.get("result").unique_id == "UC_x5XG1OV2P6uZZ5FSM9Ttw"
```

Comment:
Since the `in` check is present on the previous line, this can just use the shorter syntax
```suggestion
    assert result["result"].unique_id == "UC_x5XG1OV2P6uZZ5FSM9Ttw"
```